### PR TITLE
Changed models.py to store a string for recommendations and use produ…

### PR DESCRIPTION
**Pull request by cfnyu**
This PR has the following changes:

1. Allows for a string to be stored in the models. This is intended to hold our recommendation object as a string
2. Updated the primary key from an increasing number to the product id of the product requesting the recommendation

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -26,33 +26,36 @@ class Recommendation(object):
     """
     lock = threading.Lock()
     data = []
```
Comment (sfloam) [inline models.py:28]: here you have data as a list []

later on you have it as a dict

File: models.py [hunk 2]
```diff
@@ -63,31 +66,26 @@ def deserialize(self, data):
         """
         if not isinstance(data, dict):
             raise DataValidationError('Invalid product: body of request contained bad or no data')
+
         if data.has_key('id'):
             self.id = data['id']
+
```
Comment (sfloam) [inline models.py:72]: here you have data as a dict, earlier you had it as a list

File: models.py [hunk 3]
```diff
@@ -26,33 +26,36 @@ class Recommendation(object):
...
+        self.id = pid
+        self.recommendation = rec
 
     def save(self):
         """
         Saves a Recommendation to the data store
         """
-        if self.id == 0:
-            self.id = self.__next_index()
+
```
Comment (sfloam) [inline models.py:39]: if data is a dict, you could use keys() (which would be id)

**Review by sfloam — COMMENTED** [review]
I'm willing to approve this once you clarify the data property. We have it as a dict in one place and a list in another.

**Comment by cfnyu** [comment]
So I used the professors flask-tdd app as a starting point and it’s how he’s using it in his models.py. I’m confident this will change when we start implementing a database.

**Comment by Harivadhana** [comment]
Looks good! Thank you for discussing it in pair programming session.

**Pull request merged**